Frontend/ui-driver/src/Screens/DriverEarningsScreen/View.purs

```diff
@@ -0,0 +1,1583 @@
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        ][ textView (
+            [ height WRAP_CONTENT
+            , width $ V (((screenWidth unit - 10) * 3)/5)
+            , text item.key
+            , color Color.black900
+            , padding $ if index == 0 then Padding 4 4 0 0 else Padding 4 8 0 0 
+            ] <> if index == 0 then FontStyle.h2 TypoGraphy else FontStyle.subHeading2 TypoGraphy)
+          , linearLayout
+            [ height $ V 40
+            , width $ V 1
+            , color Color.grey900
+            , padding $ PaddingLeft 1
+            ][]
+          , linearLayout
+            [ height WRAP_CONTENT
+            , width $ V (((screenWidth unit - 10) * 2)/5)
+            , padding $ if index == 0 then Padding 4 4 0 0 else Padding 4 8 0 0 
+            ][ textView (
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text item.value
+              , color Color.black900
+              ] <> if index == 0 then FontStyle.h2 TypoGraphy else FontStyle.subHeading2 TypoGraphy)
+            , imageView $
+              [ width (V 20)
+              , height (V 20)
+              , imageWithFallback $ fetchImage FF_ASSET "ny_ic_yatri_coin"
+              , padding $ Padding 4 8 0 0 
+              , gravity CENTER_VERTICAL
+              , visibility if index == 0 then VISIBLE else GONE
+              ]
+            ]
+        ]
+      , separatorView (if (index == length tableData - 1) then false else true) state.props.subView
+    ]
+
+faqVideoView :: forall w . (Action -> Effect Unit) -> ST.DriverEarningsScreenState -> PrestoDOM (Effect Unit) w
+faqVideoView push state = 
+  if state.props.individualQuestion.videoLink /= Nothing
+    then linearLayout
+          [ width MATCH_PARENT
+          , height MATCH_PARENT
+          , background Color.red600 
+          , gravity CENTER
+          , margin $ MarginBottom 4
+          , id (getNewIDWithTag "faqVideo")
+          , afterRender
+              ( \action ->  pure $ runFn3 setYoutubePlayer (youtubeData state "VIDEO") (getNewIDWithTag "faqVideo") (show ST.PLAY))
+              (const NoAction)
+          ][]
+      else linearLayout [][]
+
+youtubeData :: ST.DriverEarningsScreenState -> String -> YoutubeData
+youtubeData state mediaType =
+  { videoTitle: "title"
+  , setVideoTitle: false
+  , showMenuButton: false
+  , showDuration: true
+  , showSeekBar: true
+  -- , videoId: getVideoID state.props.individualQuestion.videoLink
+  , videoId : maybe "" (\x -> getVideoID x) state.props.individualQuestion.videoLink
+  , videoType: "VIDEO"
+  , videoHeight: 200
+  }
+
+answersListView :: forall w . (Action -> Effect Unit) -> Array String -> ST.DriverEarningsScreenState -> PrestoDOM (Effect Unit) w
+answersListView push answersList state = 
+  linearLayout
+    [ height MATCH_PARENT
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    ] $ [faqVideoView push state] <> if length answersList > 1 
```

Comment:
refactor this as follows: 
`(map (\item -> 
        if length answersList > 1 then 
            singleAnswerView push item state
            else multipleAnswerView push item state) answersList)`